homeassistant/components/fan/icons.json

```diff
@@ -11,6 +11,17 @@
           "state": {
             "reverse": "mdi:rotate-left"
           }
+        },
+        "preset_mode": {
+          "default": "mdi:circle-medium",
+          "state": {
+            "auto": "mdi:fan-auto",
+            "boost": "mdi:rocket-launch",
+            "breeze": "mdi:weather-windy",
+            "eco": "mdi:leaf",
+            "sleep": "mdi:bed",
+            "smart": "mdi:brain"
+          }
```

Comment:
Have these been supported in the past?

Reply:
No, I don't think there were any existing icons for fan preset modes.

Reply:
I'm not sure if we should add them in that case. The ones from climate have been ported from the frontend logic (it was existing logic before we introduced these translations).

In the end, every integration can provide their own for these, the presets aren't standardized either (it is a free4all list of strings).

Reply:
There is no built-it preset. We can add the default icon in the entity_component and others in the demo platform.

Reply:
Ok, I'll leave the default and move the others to the demo platform.

There are existing fan integrations with presets. Should I add icon translations for those or submit individual PRs for each integration?

Finally, what should be done with regards to ESPHome? HA documentation states "...not possible to provide icons for state attributes using the icon property. " so there doesn't appear to be an obvious way to provided icons from the ESPHome node itself.

Reply:
> Finally, what should be done with regards to ESPHome? 

Nothing. ESPhome doesn't support this.

Reply:
Doesn't support what? Icons translation?

Reply:
The ESPHome integration does support it, but there is no way for devices to provide it.
(Different level of responsibility).

Thus: ESPHome devices will not be able to provide icons for their presets.

Reply:
Right. Personally, I think it'd be nice to provide a reasonable set of default icons (e.g. auto, sleep, eco,, etc) for ESPHome but that seems to be a no-go.

Reply:
I think that would be nice as well. Not just for ESPHome, but for MQTT as well. At this point, such construct isn't available though.

Reply:
Is that not how this would operate? Wouldn't it provides a default that integrations can override with their own icons.json?

Reply:
Your are mixing up scopes. You talk about providing them on a device level, however, this functionality is not available there.